lib/puppet/provider/zabbix_proxy/ruby.rb

```diff
@@ -60,12 +61,46 @@ def create
   end
 
   def exists?
-    check_proxy(@resource[:hostname])
+    @property_hash[:ensure] == :present
   end
 
   def destroy
     zbx.proxies.delete([zbx.proxies.get_id(host: @resource[:hostname])].flatten)
   end
 
+  #
+  # zabbix_proxy properties
+  #
   mk_resource_methods
+
+  # We have to update all parameter of the interface and mode because
+  # it might not be there if we switch the proxy from active to passive.
+  def update_mode_interface
```

Comment:
If you made this method `def flush` instead, would you even need the property setters?